### DISCUSSION TITLE
Issues when using gcc to compile CasADi's external function

### DISCUSSION THREAD

**Discussion by miyilin** [casadi/casadi#3381] [Q&A] [answered]
Hi, 

In the User Guide, there is a command "gcc −fPIC −shared gen.c −o gen.so", when I run this using my own *.c file, I got the following error
![image](https://github.com/casadi/casadi/assets/12438744/5814af85-9635-4a0e-bd98-872a95b77342)

can somebody help me with this? 

Thanks a lot!

**Comment by jgillis** [comment, marked as answer]
That instruction is a starting point, not absolute truth. Since you seem to include simulink parts, it's probably easier to use the matlab mex command to work compile it for you, or work out the flags needed for gcc. In the latter approach, perform mex with '-v' argument.

**Reply by miyilin** [reply]
I only know that mex command can generate the s-function for simulink. Since I want to use the function in the matlab_function module in simulink, an s-function may be not suitable.

The '-v' argument works.  It tells me to add the option "-DMATLAB_MEX_FILE", then the '*.so' can be generated. Thanks a lot.
